lib/SILGen/SILGenExpr.cpp

```diff
@@ -2090,8 +2090,11 @@ RValue RValueEmitter::visitFunctionConversionExpr(FunctionConversionExpr *e,
   if (srcRepTy != srcTy)
     result = convertFunctionRepresentation(SGF, e, result, srcRepTy, srcTy);
 
-  if (srcTy != destTy)
-    result = SGF.emitTransformedValue(e, result, srcTy, destTy);
+  if (srcTy != destTy) {
+    bool dontPostponeToNoEscapeCleanup = isa<BindOptionalExpr>(e->getSubExpr());
+    result = SGF.emitTransformedValue(e, result, srcTy, destTy, SGFContext(),
+                                      dontPostponeToNoEscapeCleanup);
+  }
 
```

Comment:
Ignore my previous comment. I'm not sure why this didn't show up in my first review. It's not clear why you want to use a double negative everywhere instead of just `postponeNoEscapeCleanup = true`.